airflow/providers/google/cloud/hooks/compute_ssh.py

```diff
@@ -0,0 +1,327 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import pprint
+import subprocess
+import time
+import uuid
+from io import StringIO
+from typing import Any, List, Optional, Tuple, Union
+
+import paramiko
+from google.cloud.oslogin_v1 import OsLoginServiceClient
+from googleapiclient.discovery import build
+from paramiko import SSHException
+
+from airflow import AirflowException
+from airflow.providers.google.common.hooks.base_google import GoogleBaseHook
+
+
+class ComputeEngineSshHook(GoogleBaseHook):
+    """
+    Hook to connect to a remote instance in compute engine
+
+    :param api_version: The Compute Engine discovery API version
+    :type api_version: str
+    :param gcp_conn_id: The connection id to use when fetching connection info
+    :type gcp_conn_id: str
+    :param delegate_to: The account to impersonate, if any.
+        For this to work, the service account making the request must have
+        domain-wide delegation enabled.
+    :type delegate_to: str
+    """
+    _oslogin_conn: Optional[OsLoginServiceClient] = None
```

Comment:
This creates a class attribute, not an instance attribute
https://mypy.readthedocs.io/en/stable/class_basics.html